staging/src/k8s.io/kubectl/pkg/cmd/annotate/annotate.go

```diff
@@ -238,31 +268,31 @@ func (o AnnotateOptions) Validate() error {
 		if len(o.resources) > 0 {
 			return fmt.Errorf("can only use local files by -f rsrc.yaml or --filename=rsrc.json when --local=true is set")
 		}
-		if cmdutil.IsFilenameSliceEmpty(o.Filenames, o.Kustomize) {
+		if cmdutil.IsFilenameSliceEmpty(flags.FilenameOptions.Filenames, flags.FilenameOptions.Kustomize) {
 			return fmt.Errorf("one or more files must be specified as -f rsrc.yaml or --filename=rsrc.json")
 		}
 	}
-	if len(o.newAnnotations) < 1 && len(o.removeAnnotations) < 1 && !o.list {
+	if len(o.newAnnotations) < 1 && len(o.removeAnnotations) < 1 && !flags.List {
 		return fmt.Errorf("at least one annotation update is required")
 	}
 	return validateAnnotations(o.removeAnnotations, o.newAnnotations)
 }
 
 // RunAnnotate does the work
-func (o AnnotateOptions) RunAnnotate() error {
+func (o AnnotateOptions) RunAnnotate(flags *AnnotateFlags) error {
```

Comment:
It is also a good time to rename that function to `Run`

Reply:
My first feeling was the same, somehow I don't like it. But is it really that bad? 

Things like `Local`, `Selector`, `FieldSelector`, `FileNameOptions` and some more are not available in Options. I mean, I could move them to the Options Struct again, but they are flags aren't they? 

I could also have them in both options and flags struct, like for `allnamespace`. I cannot say what is better. I am probably overlooking something.

Reply:
> Things like Local, Selector, FieldSelector, FileNameOptions and some more are not available in Options. I mean, I could move them to the Options Struct again, but they are flags aren't they?

Yes, they are definitely flag. But why we are not storing both objects in also flags and options and map them during `toOptions`?. I think there is no problem with having same fields in flag and option. As well as we have a clear simple command flow.

I understand your concern about managing same fields in different places. But if we pass `flags` into run function, in the future, PR owners will not bother themselves adding something into options, because they can just add flag and use it in `Run` function and inevitably options will become redundant.

@soltysh WDYT?